daffodil-core/src/main/scala/org/apache/daffodil/dsom/SchemaComponent.scala

```diff
@@ -54,11 +54,12 @@ trait SchemaComponent
   with PropTypes {
 
   def xml: Node
-  def parent: SchemaComponent
+  def lexicalParent: SchemaComponent
+  final lazy val optLexicalParent = Option(lexicalParent)
 
-  final override def oolagContextViaArgs = Some(parent)
+  override def oolagContextViaArgs = Option(lexicalParent)
```

Comment:
Is there a reason for using Option() instead of Some() here?

I believe the only functional difference is that Option(null) == None. If that is important here, it should be commented.

Reply:
Yup. Will add comment. That's exactly the behavior we seek.

Reply:
I will be pushing a new PR later that gets rid of the null/non-null for parents entirely. I'm sick of getting NPEs.